nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py

```diff
@@ -141,3 +142,51 @@ def __exit__(self, *args):
 
     def __del__(self):
         self.vc.release()
+
+
+class MovieDataChunkIterator(GenericDataChunkIterator):
+    """DataChunkIterator specifically for use on RecordingExtractor objects."""
+
+    def __init__(
+        self,
+        movie_file: PathType,
+        buffer_gb: float = None,
+        buffer_shape: tuple = None,
+        chunk_mb: float = None,
+        chunk_shape: tuple = None,
+        stub: bool = False,
+    ):
+        self.video_capture_ob = VideoCaptureContext(movie_file)
+        if stub:
+            self.video_capture_ob.frame_count = 10
+        self._default_chunk_shape = False
+        if chunk_shape is None:
+            chunk_shape = (1, *self.video_capture_ob.get_frame_shape())
+            self._default_chunk_shape = True
```

Comment:
```suggestion
```
I do apologize for any confusion on this, it's the most frustrating part of any DataChunkIterator class.

The `chunk_shape` and `chunk_size` refer to the actual chunking method of the `h5py.Dataset`, so this shape is mostly about streamlining read/write into and out of that object.

The `buffer_shape` and `buffer_size` arguments refer to how we actually load data from the file into RAM; the selections passed into `_get_data` are precisely these.

For this particular Iterator, I would strongly recommend overriding the default `buffer_shape` with one that fits as many full frames into RAM as possible up to the `buffer_size`. If a manual `buffer_shape` is specified it should also assert that the shape does not subset the frame size (which basically leaves the user with one free parameter - the number of frames).

Actually, that means we probably shouldn't even accept `buffer_shape` as an argument here and instead maybe just accept `buffer_size` instead since it would then be one-to-one with the number of frames.

Now, as for default chunk shape... chunk sizes should be right around or below 1MB in size. That's a roughly a single frame of a 577 x 577 RGB video and while that's in the ballpark for the historical data we prototype, on modern cameras could easily go over that. So I would add some logic for dividing up the frame shape in that situation so that each chunk stays below 1MB.

Reply:
> So I would add some logic for dividing up the frame shape in that situation so that each chunk stays below 1MB.

What is the reason to keep chunks below 1MB? 
Buffer size I understand, we want to default to a reasonable value to be RAM friendly.

Reply:
See [this part](https://github.com/hdmf-dev/hdmf/blob/dev/src/hdmf/data_utils.py#L189-L191) of the documentation for the `GenericDataChunkIterator` and the report by the HDF5 group.